erpnext/accounts/doctype/pricing_rule/pricing_rule.py

```diff
@@ -440,14 +442,19 @@ def apply_price_discount_rule(pricing_rule, item_details, args):
 		if pricing_rule.currency == args.currency:
 			pricing_rule_rate = pricing_rule.rate
 
+		# TODO https://github.com/frappe/erpnext/pull/23636 solve this in some other way.
 		if pricing_rule_rate:
 			# Override already set price list rate (from item price)
 			# if pricing_rule_rate > 0
-			item_details.update(
-				{
-					"price_list_rate": pricing_rule_rate * args.get("conversion_factor", 1),
```

Comment:
So the conversion factor coming from the args should be the `rule_conversion_factor` itself. Why this is not coming investigating and fixing that would be a proper fix